contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -437,7 +437,7 @@ func (q *queuer) Run(done <-chan struct{}) {
 			}
 
 			pod := p.(*Pod)
-			if recoverAssignedSlave(pod.Pod) != "" {
+			if recoverAssignedSlave(pod.Pod) != "" || pod.InGracefulTermination() {
```

Comment:
part of being `InGracefulTermination` is that the pod must be in
`PodRunning` phase. Is this phase possible if we're unable to recover the
assigned slave from the pod? If it's not, then how does it serve us to
check for graceful termination status here?

On Tue, Sep 22, 2015 at 9:37 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/scheduler/plugin.go
> <https://github.com/kubernetes/kubernetes/pull/13966#discussion_r40086121>
> :
>
> > @@ -437,7 +437,7 @@ func (q *queuer) Run(done <-chan struct{}) {
> >  			}
> >
> >  			pod := p.(*Pod)
> > -			if recoverAssignedSlave(pod.Pod) != "" {
> > +			if recoverAssignedSlave(pod.Pod) != "" || pod.InGracefulTermination() {
>
> The apiserver checks the NodeName. If it is empty, the deletion is
> immediate. If it is set, the graceful timeout will apply. In the case the
> NodeName is set to a non-existing node, the pod will countdown its graceful
> deletion timeout even in upstream.
>
> We have the special case that we schedule pods with NodeName != "". Even
> though this case only happens for pre-scheduled pods which are gracefully
> deleted, I would leave the above predicate. I will add a comment.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13966/files#r40086121>.
>